primo/utils/opt_utils.py

```diff
@@ -211,3 +216,46 @@ def budget_slack_variable_scaling(
         ) / model_inputs.budget
         budget_sufficient = True
     return scaling_budget_slack, budget_sufficient
+
+
+class OptimizationException(Exception):
+    """Exception for the optimization"""
+
+
+def optimization_results_handler(results: SolverResults, model: pyo.ConcreteModel):
+    """
+    A function to handle the various results returned by an optimization solve
+
+    Parameters
+    ----------
+    results : SolverResults
+        Result of an optimization solve
+
+    model : pyo.ConcreteModel
+        Pyomo model being solved
+
+    Returns
+    -------
+        1 : Optimal termination
+        2 : Feasible sub-optimal termination
+
+    Raises
+    ------
+    Exception
+        if the solver doesn't optimally terminate, raise the error
+    """
+    if check_optimal_termination(results):
+        return 1
+
+    # feasible solution - couldn't find a suitable test
+    if (
+        results.solver.status == SolverStatus.ok
+        and not results.solver.termination_condition == TerminationCondition.other
```

Comment:
Are we sure this is safe? Pareto implemented an explicit feasibility checker since solvers like CBC, Highs in general cannot be trusted. https://github.com/project-pareto/project-pareto/blob/b034951418c28d78679ce7a8759337fd7066b329/pareto/utilities/results.py#L2923